security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -253,11 +278,60 @@ func NewSecretController(ca certificateAuthority, enableNamespacesByDefault bool
 			UpdateFunc: c.namespaceUpdated,
 		})
 
+	// Extra trust anchors related routines. Holding the lock isn't strictly necessary since
+	// the controller hasn't started yet. It's added for completeness.
+	c.extraTrustAnchorMu.Lock()
+	c.lastPrimaryTrustAnchorCheck = time.Now()
+	c.extraTrustAnchors = map[string]map[string]string{}
+	c.refreshTrustAnchorsBundle()
+	c.extraTrustAnchorMu.Unlock()
+
+	extraTrustAnchorsSelector := labels.SelectorFromSet(map[string]string{ExtraTrustAnchorsLabel: "true"}).String()
+	extraTrustAnchorsLW := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			options.LabelSelector = extraTrustAnchorsSelector
+			return core.ConfigMaps(istioCaStorageNamespace).List(options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.LabelSelector = extraTrustAnchorsSelector
+			return core.ConfigMaps(istioCaStorageNamespace).Watch(options)
+		},
+	}
+	_, c.extraTrustAnchorsController = cache.NewInformer(extraTrustAnchorsLW, &v1.ConfigMap{}, time.Minute, cache.ResourceEventHandlerFuncs{
+		AddFunc: func(newObj interface{}) {
+			newConfigMap := newObj.(*v1.ConfigMap)
+
+			log.Infof("Notified about a new ConfigMap (%v) containing extra trust anchors",
+				newConfigMap.GetName())
+			c.updateExtraTrustAnchor(newConfigMap.GetName(), newConfigMap.Data)
+		},
+		DeleteFunc: func(oldObj interface{}) {
+			oldConfigMap := oldObj.(*v1.ConfigMap)
+
+			log.Infof("Notified about a deletion of an existing ConfigMap (%v) containing extra trust anchors",
+				oldConfigMap.GetName())
+			c.removeExtraTrustAnchor(oldConfigMap.GetName())
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			oldConfigMap := oldObj.(*v1.ConfigMap)
+			newConfigMap := newObj.(*v1.ConfigMap)
+
+			if oldConfigMap.ResourceVersion != newConfigMap.ResourceVersion {
+				log.Infof("Notified about an update to an existing ConfigMap (%v) containing extra trust anchors",
+					oldConfigMap.GetName())
+				c.updateExtraTrustAnchor(newConfigMap.GetName(), newConfigMap.Data)
+			}
+		},
+	})
```

Comment:
I don't feel this code should live here. Can we move it to configmap:
https://github.com/istio/istio/tree/master/security/pkg/k8s/configmap

Reply:
Done. 

FWIW, I don't think its necessarily wrong for a single controller to watch multiple resources types. It did simplify the code a bit.